app/screens/manage_channel_members/manage_channel_members.tsx

```diff
@@ -0,0 +1,277 @@
+            clearTimeout(searchTimeoutId.current);
+        }
+
+        searchTimeoutId.current = setTimeout(() => {
+            searchUsers(text);
+        }, General.SEARCH_TIMEOUT_MILLISECONDS);
+    }, [searchUsers, clearSearch]);
+
+    const updateNavigationButtons = useCallback(async (manage: boolean) => {
+        setButtons(componentId, {
+            rightButtons: [{
+                color: theme.sidebarHeaderTextColor,
+                enabled: true,
+                id: MANAGE_BUTTON,
+                showAsAction: 'always',
+                testID: 'manage_members.button',
+                text: formatMessage(manage ? messages.button_done : messages.button_manage),
+            }],
+        });
+    }, [theme.sidebarHeaderTextColor]);
+
+    const toggleManageEnabled = useCallback(() => {
+        updateNavigationButtons(!isManageMode);
+        setIsManageMode((prev) => !prev);
+    }, [isManageMode, updateNavigationButtons]);
+
+    const handleRemoveUser = useCallback(async (userId: string) => {
+        const pIndex = profiles.findIndex((user) => user.id === userId);
+        const mIndex = channelMembers.findIndex((m) => m.user_id === userId);
+        if (pIndex !== -1) {
+            const newProfiles = [...profiles];
+            newProfiles.splice(pIndex, 1);
+            setProfiles(newProfiles);
+
+            const newMembers = [...channelMembers];
+            newMembers.splice(mIndex, 1);
+            setChannelMembers(newMembers);
+
+            await NavigationStore.waitUntilScreensIsRemoved(USER_PROFILE);
+            showRemoveChannelUserSnackbar();
+        }
+    }, [profiles, channelMembers]);
+
+    const handleUserChangeRole = useCallback(async ({userId, schemeAdmin}: {userId: string; schemeAdmin: boolean}) => {
+        const clone = channelMembers.map((m) => {
+            if (m.user_id === userId) {
+                m.scheme_admin = schemeAdmin;
+                return m;
+            }
+            return m;
+        });
+
+        setChannelMembers(clone);
+    }, [channelMembers]);
+
+    const data = useMemo(() => {
+        const isSearch = Boolean(term);
+        if (isSearch) {
+            return filterProfilesMatchingTerm(searchResults, term);
+        }
+        return profiles;
+    }, [term, searchResults, profiles]);
+
+    useNavButtonPressed(MANAGE_BUTTON, componentId, toggleManageEnabled, [toggleManageEnabled]);
+
+    useEffect(() => {
+        mounted.current = true;
+        if (canManageAndRemoveMembers) {
+            updateNavigationButtons(false);
+        }
+        getProfiles();
+        return () => {
+            mounted.current = false;
+        };
+    }, [canManageAndRemoveMembers]);
+
+    useEffect(() => {
+        const removeUserListener = DeviceEventEmitter.addListener(Events.REMOVE_USER_FROM_CHANNEL, handleRemoveUser);
+        const changeUserRoleListener = DeviceEventEmitter.addListener(Events.MANAGE_USER_CHANGE_ROLE, handleUserChangeRole);
```

Comment:
Curious why we used the event route here.

Reply:
The main reason is to reduce prop drilling and passing `handleRemoveUser` and `handleUserChangeRole`.  The list of users does not come from the database and instead comes from a static set of values from the server.  The profile bottom screen is many components deep from this parent component.

Reply:
Can be left like this, but just wonder if using a context would have been a more "react-like" way to solve this.

I am not a big fan of events because it is a bit harder to track down who is listening, who is sending, etc...
Context is also not my cup of tea, since it is also complicated to tie down who is the provider supplying this component.

Both add some "hidden element" to the code that I am not a big fan of, but at least the context is more "react-like". (Sure, events are very JS-like, so not a bad option neither...)

Either way, all this ranting to say... LGTM :P

Reply:
Then the text editor could find all references of `EventEmitterManger.RemoveUser`

Reply:
The react way of doing these things is using a context. We have that in a few places, so if you are curious, you can take a look at `app/context/`. If you want to see the actual react documentation, this may be useful: https://beta.reactjs.org/reference/react/useContext

Regarding what is better / more performant... haven't looked into it :P

Reply:
Thanks!